src/org/exist/security/internal/SecurityManagerImpl.java

```diff
@@ -753,15 +745,41 @@ private Realm findRealmForRealmId(final String realmId) throws ConfigurationExce
         }
         throw new ConfigurationException("Realm id = '" + realmId + "' not found.");
     }
-    
-    @Override
-    public void addGroup(final int id, final Group group) {
-        groupsById.modify(principalDb -> principalDb.put(id, group));
+
+    /**
+     * Register mapping id to group.
+     *
+     * @param group
+     */
+    public void registerGroup(final Group group) {
+        groupsById.modify(principalDb -> {
+
+            final int id = group.getId();
+
+            principalDb.put(id, group);
+
+            if (id < MAX_GROUP_ID) {
+                lastGroupId = Math.max(lastGroupId, id);
+            }
+        });
     }
 
-    @Override
-    public void addUser(final int id, final Account account) {
-        usersById.modify(principalDb -> principalDb.put(id, account));
+    /**
+     * Register mapping id to account.
+     *
+     * @param account
+     */
+    public void registerAccount(final Account account) {
+        usersById.modify(principalDb -> {
+
+            final int id = account.getId();
+
+            principalDb.put(id, account);
+
+            if (id < MAX_USER_ID) {
+                lastAccountId = Math.max(lastAccountId, id);
```

Comment:
I can't quite put my finger on it, but taking the Math.max here seems to be either unnecessary or a guard against some other problem which should likely be fixed at the root cause instead.

Can you detail a use-case where `id` might be bigger than `lastAccountId`?